Docs: add delegation use-case and clean up existing use-cases (#3138)
input-output-hk/cardano-wallet: docs/user-guide/common-use-cases/pab.md

 - In order to be able to balance Plutus transaction we need funds on the wallet. In case of [Testnet](https://testnets.cardano.org/en/testnets/cardano/overview/) we can request tADA from the [faucet](https://testnets.cardano.org/en/testnets/cardano/tools/faucet/).

## Overview
This guide is to show how to invoke Plutus contracts with cardano-wallet.

## Workflow
Once you have created a smart contract with [PAB](https://github.com/input-output-hk/plutus/tree/master/plutus-pab) you can execute it via cardano-wallet.

There are three endpoints that need to be invoked to follow the workflow:

 - [POST /wallets/{walletId}/transactions-balance](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/balanceTransaction) - for balancing transaction from PAB.
 - [POST /wallets/{walletId}/transactions-sign](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/signTransaction) - for signing transaction.
 - [POST   
/wallets/{walletId}/transactions-submit](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/submitTransaction) - for submitting transaction to the network.

### Balance transaction

> Plutus Application Backend provides payload of unbalanced transaction. This transaction needs to be balanced with wallet inputs such that it can be submitted to the network. The response from this endpoint returns `fee` and `coin_selection` of balanced transaction as well as CBOR-encoded `transaction` represented in base64 encoding. We will need that returned `transaction` value to pass on to [POST /wallets/{walletId}/transactions-sign](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/signTransaction) endpoint.

Reviewer: ```suggestion
Plutus Application Backend provides a payload of an unbalanced transaction. This transaction needs to be balanced with wallet's inputs such that it can be submitted to the network. The response from this endpoint returns `fee` and `coin_selection` of the balanced transaction as well as CBOR-encoded `transaction` represented in base64 encoding. We will need the returned `transaction` value to pass on to [POST /wallets/{walletId}/transactions-sign](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/signTransaction) endpoint.
```

### Sign transaction

> Once transaction is balanced we need to sign it using our wallet's secure passphrase passing previously returned CBOR-encoded `transaction`. The sign endpoint again will returned CBOR-encoded `transaction` which need to pass further to [POST   

Reviewer: ```suggestion
Once the transaction is balanced we need to sign it using our wallet's secure passphrase and pass the previously returned CBOR-encoded `transaction`. The sign endpoint will again return CBOR-encoded `transaction` which is needed to be passed further to [POST   
```

### Submit transaction

We have our balanced and signed CBOR-encoded `transaction` represented in base64 encoding. Now we can submit it to the network with [POST   
/wallets/{walletId}/transactions-submit](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/submitTransaction) endpoint.

```
$ curl -X POST http://localhost:8090/v2/wallets/1f82e83772b7579fc0854bd13db6a9cce21ccd95/transactions-submit \  
-d '{"transaction":"hKYAgYJYIAQh6enKOwhicVBXDNo6ukKgvJJnMVFIbrvczELZao8sAQ2BglggBCHp6co7CGJxUFcM2jq6QqC8kmcxUUhuu9zMQtlqjywBAYKCWDkA87i32rzLSeZJCB3GHrOB/Shg/B+3I7ucILBfkyiKiq2biY2B6ZIS5v5pD/BMyrCeg3/pX6imiDEaBoFcr4NYHXBNcs9WmjOaGKfZMCMTmD9W4Nls1FvcsdZRLcpqGgAehIBYIJI5GOQDv0PDS072tI6y7gS6vtFzINjRuf+a0Ibob0TsAhoAEdGtDoALWCAvUOolRvjOAgykW/zyq+sC/xivIoNGb4iK5IkYSz0tOaIAgYJYIBaZLkF/espB4YMd23QgZlpQ0TQkXR8ESalTVqlS4HFQWEAUwu5e080HmdIBoPAeb+L22gFtvlY7j8urDPoRIqxb0d41LF/MWNYXQy9yahWdqyHzCg+P7mduRuGrXAFUR0EBBIHYeYD19g=="}' \  
-H "Content-Type: application/json"

{"id":"148cb83bb2943e54d878a86fdc4309d952ca5fd62b3463fbb72a94180ebe8e82"}
```

> We can monitor status of submitted transaction using [GET /wallets/{walletId}/transactions/{transactionId}](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getTransaction) endpoint.

Reviewer: ```suggestion
We can monitor status of the submitted transaction using [GET /wallets/{walletId}/transactions/{transactionId}](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getTransaction) endpoint.
```
